pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -395,3 +429,53 @@ func (c *sshClient) runSSH(command string) Output {
 		Err:    err,
 	}
 }
+
+func removeUselessLinesFromCloudInitOutput(s string) string {
+	regexes := []string{
+		`^\s+\d+K [. ]+ \d+%.*\ds$`,                  //  10000K .......... .......... .......... .......... ..........  6%!M(MISSING) 1s
+		`^Get:\d+ http.*ubuntu.com.*kB.*`,            // Get:17 http://archive.ubuntu.com/ubuntu focal/universe Translation-en [5,124 kB[]`
+		`^Preparing to unpack \.\.\..*`,              // Preparing to unpack .../04-libx11-6_2%!a(MISSING)1.6.9-2ubuntu1.6_amd64.deb ...\r
+		`^Selecting previously unselected package.*`, // Selecting previously unselected package kubeadm.\r
+		`^Setting up .* \.\.\..*`,                    // Setting up hicolor-icon-theme (0.17-2) ...\r
+		`^Unpacking .* \.\.\..*`,                     // Unpacking libatk1.0-0:amd64 (2.35.1-1ubuntu2) ...\r
+	}
+
+	// Compile the regexes
+	var compiledRegexes []*regexp.Regexp
+	for _, re := range regexes {
+		compiled, err := regexp.Compile(re)
+		if err != nil {
+			msg := fmt.Sprintf("removeUselessLinesFromCloudInitOutput: Failed to compile regex %s: %v\n", re, err)
+			fmt.Println(msg)
+			return msg
+		}
+		compiledRegexes = append(compiledRegexes, compiled)
+	}
+
+	var new []string
+
+	scanner := bufio.NewScanner(strings.NewReader(s))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		// Check if the line matches any of the regexes
+		matches := false
+		for _, re := range compiledRegexes {
+			if re.MatchString(line) {
+				matches = true
+				break
+			}
+		}
+
+		// If the line does not match any regex, print it
+		if matches {
+			continue
+		}
+		new = append(new, line)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return fmt.Sprintf("Error reading string: %v\n", err)
```

Comment:
I am happy with this. If there is an error at this level an event containing an error message is fine. 

It is very unlikely that an error occurs here, since no network or disk IO is happening here. It is just in memory (reading a string).